crypto/provider_core.c

```diff
@@ -1111,7 +1111,9 @@ static int provider_deactivate(OSSL_PROVIDER *prov, int upcalls,
         return -1;
     }
 
-    CRYPTO_atomic_add(&prov->activatecnt, -1, &count, prov->activatecnt_lock);
+    if (!CRYPTO_atomic_add(&prov->activatecnt, -1, &count, prov->activatecnt_lock))
+        return -1;
```

Comment:
If we do this then we need to unlock the `prov->flag_lock` and `store->lock` if `lock` is set.

Reply:
Knew I missed something, thank you!